sherpa/astro/ui/tests/test_astro_session.py

```diff
@@ -406,3 +413,815 @@ def test_save_error_ascii(session, kwargs, idval, tmp_path):
+@requires_fits  # only for AstroSession, but not worth being clever
+@pytest.mark.parametrize("session", [Session, AstroSession])
+@pytest.mark.parametrize("label", ["model", "source"])
+def test_get_xxx_component_plot_with_templates_recalc_false_no_interp(session, label, make_data_path):
+    """What is the intended behavior for template models?
+
+    This is a regresion test.
+
+    """
+
+    # This sets up a template model for the default id, so try
+    # calling with this model but a different dataset.
+    #
+    s = session()
+    dname = make_data_path('load_template_with_interpolation-bb_data.dat')
+    s.load_data(dname)
+
+    bbtemp, ynorm = setup_template_model(s, make_data_path, interp=None)
+    s.set_source(ynorm * bbtemp)
+
+    # The plot results should be empty
+    #
+    mplot = getattr(s, f"get_{label}_plot")(2, recalc=False)
+    cplot = getattr(s, f"get_{label}_component_plot")(2, bbtemp, recalc=False)
+
+    mclass = getattr(sherpa.plot, f"{label.capitalize()}Plot")
+    assert isinstance(mplot, mclass)
+
+    cclass = getattr(sherpa.plot, f"Component{label.capitalize()}Plot")
+    assert isinstance(cplot, cclass)
+
+    # The logic here is not-at-all obvious, but just test the current
+    # behavior.
+    #
+    subclass = getattr(sherpa.plot, f"ComponentTemplate{label.capitalize()}Plot")
+    if label == "source":
+        assert isinstance(cplot, subclass)
+    else:
+        assert not isinstance(cplot, subclass)
+
+    assert mplot.x is None
+    assert cplot.x is None
+    assert mplot.title == label.capitalize()
+    assert cplot.title == label.capitalize()
+
+
+@requires_data
+@requires_fits  # only for AstroSession, but not worth being clever
+@pytest.mark.parametrize("session", [Session, AstroSession])
+@pytest.mark.parametrize("label", ["model", "source"])
+def test_get_xxx_component_plot_with_templates_data1d(session, label, make_data_path):
+    """What is the intended behavior for template models?  Data1D
+
+    As this is not a PHA data set the source and model values should
+    be the same (although plot labels will be different).
+
+    This is a regression test as it is not obvious what the intended
+    logic is.
+
+    """
+
+    s = session()
+    dname = make_data_path('load_template_with_interpolation-bb_data.dat')
+    s.load_data(dname)
+    bbtemp, ynorm = setup_template_model(s, make_data_path)
+    s.set_source(bbtemp * ynorm)
+
+    # Because the source model is a composite, the y values of the
+    # two should differ by ynorm.
+    #
+    mplot = getattr(s, f"get_{label}_plot")()
+    cplot = getattr(s, f"get_{label}_component_plot")(bbtemp)
+
+    mclass = getattr(sherpa.plot, f"{label.capitalize()}Plot")
+    assert isinstance(mplot, mclass)
+
+    cclass = getattr(sherpa.plot, f"Component{label.capitalize()}Plot")
+    assert isinstance(cplot, cclass)
+
+    # Check it's not a sub-class.
+    #
+    subclass = getattr(sherpa.plot, f"ComponentTemplate{label.capitalize()}Plot")
```

Comment:
This is an example where it's not clear to me why this is worth testing. Why could this not be a sub-class? (No need to explain this to me, just read on and click "resolve conversation".) However, that's fine for a Regression test. It passes now, so it clearly tests existing behavior. If it ever changes, someone will have to confirm that ti's an intended change.

Reply:
It is a regression test. I don't understand the template plotting code and what it should do, so I'm just testing what it does do now and leaving it to #1530 for future discussion.